### PR TITLE
PLT-290: Added Validation for cluster, datastore, network and datacenter in day 2 operations

### PR DESCRIPTION
**Pull request by SivaanandM**
…ter in day 2 opeartions

### CODE REVIEW

File: examples/resources/spectrocloud_cluster_vsphere/resource.tf [hunk 1]
```diff
@@ -1,18 +1,19 @@
 data "spectrocloud_cluster_profile" "vmware_profile" {
   name    = "vmware-public-repo"
-  version = "2.0.0"
+  version = "1.0.0"
   context = "tenant"
 }
 data "spectrocloud_cloudaccount_vsphere" "vmware_account" {
-  name = "gm-pcg-wop-d"
+  name = var.shared_vmware_cloud_account_name
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_vsphere/resource.tf:7]: @SivaanandM could you please provide complementary change in variables.tf?

File: spectrocloud/resource_cluster_vsphere.go [hunk 2]
```diff
@@ -452,6 +455,19 @@ func resourceClusterVsphereUpdate(ctx context.Context, d *schema.ResourceData, m
 
 	if d.HasChange("machine_pool") {
 		oraw, nraw := d.GetChange("machine_pool")
+		if oraw != nil && nraw != nil {
```
Comment (nikchern) [inline spectrocloud/resource_cluster_vsphere.go:458]: Let's have a unit test to cover this update function?

File: examples/resources/spectrocloud_cluster_vsphere/resource.tf (hunk 1 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_vsphere/resource.tf:7]: usually, we won't have a custom variable defined in variable.tf, we used to maintain in tfvars. I will revert back to hard values

File: spectrocloud/resource_cluster_vsphere.go (hunk 2 above)
Comment (SivaanandM) [reply spectrocloud/resource_cluster_vsphere.go:458]: As discussed earlier, we currently don't have support to simulate state change (has change for particular attribute ) in terraform SDK through the unit test. we can't cover it in the unit test for now

**Review by nikchern — APPROVED** [review]

**Pull request merged**